test/apps/iodemo/run_io_demo.sh

```diff
@@ -0,0 +1,456 @@
+		tasks_per_node=$((${TOTAL_CPUS} / ${SLURM_NNODES}))
+	fi
+}
+
+# Set default configuration values
+init_config()
+{
+	verbose=0
+	iodemo_exe=""
+	iodemo_client_args=""
+	num_clients=1
+	num_servers=1
+	map_by="node"
+	base_port_num=20000
+	tasks_per_node=1
+	net_if="bond0"
+	duration=30
+
+	# command line args will override slurm env vars
+	check_slurm_env
+}
+
+# Show the configuration
+show_config()
+{
+	echo "Launch configuration:"
+	for key in \
+			host_list tasks_per_node map_by \
+			num_clients num_servers \
+			iodemo_exe iodemo_client_args \
+			net_if base_port_num \
+			duration
+	do
+		show_var ${key}
+	done
+}
+
+# Helper to show default value if defined, otherwise empty string
+show_default_value()
+{
+	value=$(eval echo \$$1)
+	[ -z "${value}" ] && echo "" || echo " (${value})"
+}
+
+# Show script usage help message
+usage()
+{
+	echo "Usage: run_io_demo.sh <options> <exe> <arguments>"
+	echo
+	echo "  -h|--help                   Show this help message"
+	echo "  -v|--verbose                Turn on verbosity"
+	echo "  -H|--hostlist <h1>,<h2>,..  List of host names to run on"$(show_default_value host_list)
+	echo "  -i|--netif <n>              Network interface to use"$(show_default_value net_if)
+	echo "  -d|--duration <seconds>     How much time to run the application"$(show_default_value duration)
+	echo "  --num-clients <count>       Number of clients to run"$(show_default_value num_clients)
+	echo "  --num-servers <count>       Number of servers to run"$(show_default_value num_servers)
+	echo "  --tasks-per-node <count>    Maximal number of tasks per node"$(show_default_value tasks_per_node)
+	echo "  --map-by <mode>             Mapping mode of processes to hosts"$(show_default_value map_by)
+	echo "                               - node : Distribute clients and servers equally"
+	echo "                               - slot : Place the servers first, then the clients"
+	echo
+}
+
+# Parse command line arguments and update configuration variables
+parse_args()
+{
+	while [[ $# -gt 0 ]]
+	do
+		key="$1"
+		case $key in
+		-h|--help)
+			usage
+			exit 0
+			;;
+		-v|--verbose)
+			verbose=1
+			;;
+		-H|--hostlist)
+			host_list="$2"
+			shift
+			;;
+		-i|--netif)
+			net_if="$2"
+			shift
+			;;
+		-d|--duration)
+			duration="$2"
+			shift
+			;;
+		--num-clients)
+			num_clients="$2"
+			shift
+			;;
+		--num-servers)
+			num_servers="$2"
+			shift
+			;;
+		--tasks-per-node)
+			tasks_per_node="$2"
+			shift
+			;;
+		--map-by)
+			map_by="$2"
+			shift
+			;;
+		[^-]*)
+			iodemo_exe="$key"
+			shift
+			break
+			;;
+		*)
+			echo "Invalid parameter '${key}'"
+			usage
+			exit -2
+			;;
+		esac
+		shift
+	done
+
+	iodemo_client_args="$@"
+}
+
+check_params()
+{
+	if [ -z ${iodemo_exe} ]
+	then
+		echo "Error: missing executable from command line"
+		exit -2
+	fi
+}
+
+collect_ip_addrs()
+{
+	# convert the output of 'ip' to 'host:ip' list
+	host_ips=$(pdsh -w ${host_list} ip -4 -o address show ${net_if} |
```

Comment:
s/pdsh/$launcher
and maybe have at the beginning smth like that (clush and pdsh have exactly same syntax)
``` bash
for xx in pdsh clush; do
   if type $xx; then
      launcher=$xx
      break
   fi
done

if [ -z "$launcher" ]; then
   exit 1
fi
```